cmd/shim/c2h.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"strings"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func tunnel(args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("usage: tunnel <socket>:<port>[/<tcp|udp>] [...]")
+	}
+
+	eg, ctx := errgroup.WithContext(context.Background())
+
+	for _, trio := range args {
+		upstreamSock, portSpec, ok := strings.Cut(trio, ":")
+		if !ok {
+			return fmt.Errorf("invalid socket:port:protocol spec: %s", trio)
+		}
+
+		port, network, ok := strings.Cut(portSpec, "/")
+		if !ok {
+			network = "tcp"
+		}
+
+		eg.Go(func() error {
+			return tunnelOne(ctx, upstreamSock, port, network)
+		})
```

Comment:
Looks like right now it'll stop them all as soon as one hits a dial error while trying to forward, which is pretty janky.

I'll just not use `errgroup.WithContext`, and change it to just continue accepting connections in the face of `Dial` errors, since it could just be that the other end couldn't accept (maybe too many open files), and might recover. With those changes the tunnel should just keep going until you stop it.

Reply:
Hm, I guess there are some cases worth bailing on. For example is `net.Listen` fails it should probably fail the whole tunnel since that would be some sort of weird misconfiguration, like listing the same port multiple times. Guess I'll keep parts but just change `net.Dial` to `continue`. And maybe `l.Accept`? :thinking:

Reply:
`l.Accept` has sent me down a rabbit hole: apparently `Temporary() bool` is deprecated now (https://github.com/golang/go/issues/45729), and there's [some regret](https://groups.google.com/g/golang-nuts/c/-JcZzOkyqYI/m/xwaZzjCgAwAJ) over having `net/http` default to retrying on `EMFILE`, so maybe I'll just let it fail and exit all the other listeners, since they'd be likely to hit `EMFILE` at the same time anyway.